### PR TITLE
Refactor addon to only apply on active weapon

### PR DESCRIPTION
**Pull request by wrefgtzweve**
This PR makes it so movementspeed is only decided by your held weapon, not by all your equipped weapons.

### CODE REVIEW

**Review by StrawWagen — DISMISSED** [review]
Looks good, works in single

File: lua/autorun/server/sv_pvp_movespeed.lua [hunk 1]
```diff
@@ -77,69 +46,55 @@ local function setSpeedFromWeight( ply, totalWeight )
...
-end
-pvpMoveSpeed.setSpeedFromWeight = setSpeedFromWeight
-
-local function isPACWeapon( weapon )
-    return string.sub( weapon:GetClass(), 1, 4 ) == "pac_"
 end
 
 local function getWeaponWeight( weapon )
-    if isPACWeapon( weapon ) then return 0 end
+    if string.sub( weapon:GetClass(), 1, 4 ) == "pac_" then return 0 end
```
Comment (sarahsturgeon) [inline lua/autorun/server/sv_pvp_movespeed.lua:57]: I prefer this as its own function for readability/maintainability reasons but I won't make a fuss about it.

We could also look at making a hook

File: lua/autorun/server/sv_pvp_movespeed.lua [hunk 2]
```diff
@@ -77,69 +46,55 @@ local function setSpeedFromWeight( ply, totalWeight )
...
-pvpMoveSpeed.getWeaponWeight = getWeaponWeight
 
 local function getPlayerWeight( ply )
     if ply.IsInBuild and ply:IsInBuild() then return 0 end
-    local weapons = ply:GetWeapons()
+    local activeWeapon = ply:GetActiveWeapon()
     local totalWeight = 0
-    for _, weapon in pairs( weapons ) do
-        totalWeight = totalWeight + getWeaponWeight( weapon )
+    if IsValid( activeWeapon ) then
```
Comment (sarahsturgeon) [inline lua/autorun/server/sv_pvp_movespeed.lua:66]: nit: Localize `IsValid` in this file?

File: lua/autorun/server/sv_pvp_movespeed.lua [hunk 3]
```diff
@@ -150,24 +105,10 @@ end
...
 end
 
-
 -- Hooks --
-hook.Remove( "WeaponEquip", generateCFCHook( "HandleEquipMS" ) )
-hook.Add( "WeaponEquip", generateCFCHook( "HandleEquipMS" ), onEquip )
-
-hook.Remove( "PlayerDroppedWeapon", generateCFCHook( "HandleDroppedWeaponMS" ) )
-hook.Add( "PlayerDroppedWeapon", generateCFCHook( "HandleDroppedWeaponMS" ), onDrop )
+hook.Remove( "PlayerSwitchWeapon", "CFC_PlyMS_PlayerSwitchWeapon" )
```
Comment (sarahsturgeon) [inline lua/autorun/server/sv_pvp_movespeed.lua]:
```suggestion
```

Leftover dev code

**Review by sarahsturgeon — DISMISSED** [review]
👍

**Review by StrawWagen — APPROVED** [review]

**Pull request merged**